### PR TITLE
soc: esp32s3: add initial board bring up

### PR DESCRIPTION
**Pull request by sylvioalves**

### CODE REVIEW

File: zephyr/esp_shared/include/stubs.h [hunk 1]
```diff
@@ -41,6 +41,10 @@
 #define CONFIG_ESP_CONSOLE_UART_NUM 2
 #define CONFIG_ESP_CONSOLE_UART_BAUDRATE DT_PROP(DT_NODELABEL(uart2), current_speed)
 
+#else
+
+#define CONFIG_ESP_CONSOLE_UART_NUM -1
```
Comment (sylvioalves) [inline zephyr/esp_shared/include/stubs.h:46]: This is missing and required. When uart is disabled, i.e., using USB interface, build will fail. This fixes it.

File: tools/cmake/toolchain-esp32s3.cmake [hunk 2]
```diff
@@ -7,3 +7,4 @@ set(CMAKE_SYSTEM_NAME Generic)
 
 set(CMAKE_C_FLAGS "-mlongcalls" CACHE STRING "C Compiler Base Flags")
 set(CMAKE_CXX_FLAGS "-mlongcalls" CACHE STRING "C++ Compiler Base Flags")
+set(CMAKE_EXE_LINKER_FLAGS "-nostartfiles -nodefaultlibs" CACHE STRING "Linker Base Flags")
```
Comment (LucasTambor) [inline tools/cmake/toolchain-esp32s3.cmake:10]: Need newline at EOF

File: zephyr/esp_shared/pincfgs/esp32s3.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Notes:
```
Comment (LucasTambor) [inline zephyr/esp_shared/pincfgs/esp32s3.yml:4]: The note above and the definitions below are not coherent. Which one is valid?

File: tools/cmake/toolchain-esp32s3.cmake (hunk 2 above)
Comment (LucasTambor) [reply tools/cmake/toolchain-esp32s3.cmake:10]: Actually there are more files that are complaining about newlines

**Comment by sylvioalves** [comment]
@LucasTambor, thanks, PTAL again. I cleaned all RTC changes. It works as is without updates for now.

**Review by LucasTambor — APPROVED** [review]

File: components/soc/esp32s3/include/soc/rtc.h [hunk 4]
```diff
@@ -44,7 +44,7 @@ extern "C" {
  * - rtc_init: initialization
  */
 
-#define MHZ (1000000)
+// #define MHZ (1000000)
```
[the remark this answers is not in the document]
Comment (sylvioalves) [reply components/soc/esp32s3/include/soc/rtc.h:47]: Yes, I'll add a check in there.

**Review by LucasTambor — APPROVED** [review]

**Comment by sylvioalves** [comment]
@marekmatej PTAL

**Review by marekmatej — APPROVED** [review]
LGTM

**Pull request merged**